### PR TITLE
Output document_id for redirection file for BI usage

### PR DESCRIPTION
**Pull request by live1206**
[AB#193606](https://dev.azure.com/ceapex/Engineering/_workitems/edit/193606)

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/docfx/pull/5766)

### CODE REVIEW

File: src/docfx/build/redirection/BuildRedirection.cs [hunk 1]
```diff
@@ -27,6 +28,11 @@ internal static List<Error> Build(Context context, Document file)
                 file.Mime.Value);
 
             publishItem.RedirectUrl = context.RedirectionProvider.GetRedirectUrl(file.FilePath);
+            var (documentId, _) = context.DocumentProvider.GetDocumentId(context.RedirectionProvider.GetOriginalFile(file.FilePath));
```
Comment (yufeih) [inline src/docfx/build/redirection/BuildRedirection.cs:31]: Do we also want to include `document_version_independent_id`? They seem to appear hand in hand. Though the bug/feature may not say it is required.

Comment (live1206) [reply src/docfx/build/redirection/BuildRedirection.cs:31]: I can sync with the requester, not required from the description

Comment (herohua) [reply src/docfx/build/redirection/BuildRedirection.cs:31]: Yes, we do.

File: docs/specs/redirection.yml [hunk 2]
```diff
@@ -88,6 +89,8 @@ inputs:
     renames:
       a.md: https://docs.microsoft.com/en-us/b?query#fragment
 outputs:
+  .publish.json: |
```
Comment (herohua) [inline docs/specs/redirection.yml:92]: also add a case that `redirect_document_id == false`?

File: src/docfx/build/redirection/BuildRedirection.cs [hunk 3]
```diff
@@ -27,6 +28,13 @@ internal static List<Error> Build(Context context, Document file)
                 file.Mime.Value);
 
             publishItem.RedirectUrl = context.RedirectionProvider.GetRedirectUrl(file.FilePath);
+            var (documentId, documentVersionIndependentId) = context.DocumentProvider.GetDocumentId(context.RedirectionProvider.GetOriginalFile(file.FilePath));
+            publishItem.ExtensionData = new JObject
+            {
+                ["document_id"] = documentId,
+                ["document_version_independent_id"] = documentVersionIndependentId,
+                ["canonical_url"] = file.CanonicalUrl,
```
Comment (yufeih) [inline src/docfx/build/redirection/BuildRedirection.cs:36]: Is `canonical_url` required?  canonical urls are for HTML page SEO purpose, redirection files doesn't get a chance to show the HTML page.

 https://yoast.com/what-is-a-canonical-url/

Comment (jesong) [reply src/docfx/build/redirection/BuildRedirection.cs:36]: BI team need to know the page url. Provide it from canonical_url or from other metadata both OK to me. But, we need the url.

File: docs/specs/redirection.yml (hunk 2 above)
Comment (live1206) [reply docs/specs/redirection.yml:92]: good point, test added

File: src/docfx/build/redirection/BuildRedirection.cs (hunk 3 above)
Comment (yufeih) [reply src/docfx/build/redirection/BuildRedirection.cs:36]: What does v2 and conceptual documents use for this info?

**Pull request merged**